File: app/src/main/java/com/zack/enderplan/view/contract/TypeEditViewContract.java
package com.zack.enderplan.view.contract;

import android.support.annotation.DrawableRes;
import android.support.annotation.StringRes;

import com.zack.enderplan.model.bean.FormattedType;

public interface TypeEditViewContract extends BaseViewContract {

    void showInitialView(FormattedType formattedType);

    void showTypeNameEditorDialog(String originalEditorText);

    void onTypeNameChanged(String typeName, String firstChar);

    void onTypeMarkColorChanged(int colorInt, String colorName);

    void onTypeMarkPatternChanged(boolean hasPattern, @DrawableRes int patternResId, String patternName);

    void showTypeMarkColorPickerDialog(String defaultColor);

    void showTypeMarkPatternPickerDialog(String defaultPattern);
}

File: app/src/main/java/net/zackzhang/app/end/view/contract/TypeEditViewContract.kt
package net.zackzhang.app.end.view.contract

import android.support.annotation.DrawableRes

import net.zackzhang.app.end.model.bean.FormattedType

interface TypeEditViewContract : BaseViewContract {

    fun showInitialView(formattedType: FormattedType)

    fun showTypeNameEditorDialog(originalEditorText: String)

    fun onTypeNameChanged(typeName: String, firstChar: String)

    fun onTypeMarkColorChanged(colorInt: Int, colorName: String)

    fun onTypeMarkPatternChanged(hasPattern: Boolean, @DrawableRes patternResId: Int, patternName: String?)

    fun showTypeMarkColorPickerDialog(defaultColor: Int)

    fun showTypeMarkPatternPickerDialog(defaultPattern: String?)
}
